src/utils.ts

```diff
@@ -49,6 +50,18 @@ export const getProtocolModuleForUri = (uri: string) => {
   return uri.indexOf('https') === 0 ? https : http;
 };
 
+export const extractEnvVars = () => {
+  const res = {};
+  let length = 0;
+  Object.entries(process.env).forEach(([key, value]) => {
+    length += value.length;
+    if (length <= MAX_SIZE) {
+      res[key] = value;
+    }
```

Comment:
I think that there is a bug here, we increase `length` before we decide to assign the value.
Please add a test similar to:
```
    it('extractEnvVars should filter exes data', () => {
      process.env = {
        a: '1'.repeat(5000),
        b: '2',
        c: '3',
      };
      const env = extractEnvVars();
      expect(env).toEqual({
        b: '2',
        c: '3',
      });
    })
```